geozero/tests/geozero-api.rs

```diff
@@ -86,20 +77,25 @@ impl PropertyProcessor for FeatureFinder {
 
 #[test]
 fn feature_finder() -> Result<()> {
-    let mut filein = BufReader::new(File::open("tests/data/countries.fgb")?);
-    let mut fgb = FgbReader::open(&mut filein)?.select_all()?;
+    let mut filein = BufReader::new(File::open("tests/data/countries.geojson")?);
+    let mut json = GeoJsonLineReader::new(&mut filein);
 
     let mut finder = FeatureFinder {};
-    while let Some(feature) = fgb.next()? {
-        let found = feature.process_properties(&mut finder);
-        if found.is_err() || found.unwrap() {
-            break;
-        }
-    }
-    let feature = fgb.cur_feature();
-    let props = feature.properties()?;
-    assert_eq!(props["id"], "DNK".to_string());
-    assert_eq!(props["name"], "Denmark".to_string());
+    // process_properties is not public. We should have iterators!
```

Comment:
Can you explain more what you mean by "we should have iterators"?

My guess is that you mean something like:

```
GeoJsonLineReader::next(&mut self) -> Result<Option<geojson::Feature>> 
```

So that we can write:

```
while let Some(feature) = json.next()? {
    ...
}
```

Is that right? Or something else? If so, that seems reasonable.

---

Though, TBH, this example doesn't seem like a great fit for the geozero API anyway. In my experience, geozero is mostly concerned with processing entire datasets. This example relies mostly on things intrinsic to flatgeobuf, including some API's added to FGB as helpers for implementing geozero APIs, but are not actually geozero APIs themselves.

Reply:
Exactly. We should have a common feature iterator for `GeozeroDatasource`.

Reply:
I'd really love to have a feature iterator. I've thought briefly about it before but I got stuck because I don't know how to combine the push-based nature of the `GeomProcessor` with the pull-based nature of the iterators